### PR TITLE
set of new dhcp options

### PR DESCRIPTION
**Pull request by wwencel**
This is just a checklist to guide you. You can remove it safely.

**Checklist:**

-   [ x ] If you are new to Scapy: I have checked <https://github.com/secdev/scapy/blob/master/CONTRIBUTING.md> (esp. section submitting-pull-requests)
-   [ x ] I squashed commits belonging together
-   [ x ] I added unit tests or explained why they are not relevant
-   [ x ] I executed the regression tests for Python2 and Python3 - haven't notice any regressions

> brief description what this PR will do, e.g. fixes broken dissection of XXX
extend list of standard options available in DHCP v4 
> if required - short explanation why you fixed something in a way that may look more complicated as it actually is
n/a
> if required - outline impacts on other parts of the library
n/a

### CODE REVIEW

**Comment by gpotter2** [comment]
Thanks for the PR !
There is a unit tests that you will need to update.

File: test/regression.uts [hunk 1]
```diff
@@ -11449,14 +11449,15 @@ assert(param1.random != param2.random)
 ############
 ############
 + DHCP
-
+~ teraz
```
Comment (gpotter2) [inline test/regression.uts]: What's this ?

File: test/regression.uts [hunk 2]
```diff
@@ -11449,14 +11449,15 @@ assert(param1.random != param2.random)
...
 = BOOTP - misc
 assert BOOTP().answers(BOOTP())
 assert BOOTP().hashret() == b"\x00\x00\x00\x00"
 
 import random
 random.seed(0x2807)
-assert str(RandDHCPOptions(size=1)) in [r"[('WWW_server', '36.35.181.183')]",
+assert str(RandDHCPOptions(size=1)) in [r"[('netinfo-server-tag', '\xe4Z\xdb\xef\xaf\xf5\x10\x91\x9eD\x0c9. \x91\xa5*Ug\x9cl\xa1\xca\xe3\xa2\xf0:\xd7\x00\xc9g\x02f1\xbd?\xb3\xc4\xe10S7\x9d\x91\x84\xc9y\xb2V<\x7f\x87\x14\xde\xc2\xf4\x94\xabyk\xc1\xbcU\x14\xcb\xb8\xcf\x96\xfedU\x81Z\xff\t\xa6\xe0\xf5\x15p\x91\xf0\xec\x7f')]",
+                                        r"[('WWW_server', '36.35.181.183')]",
                                         r"[('boot-file-name', '\xe4Z\xdb\xef\xaf\xf5\x10\x91\x9eD\x0c9. \x91\xa5*Ug\x9cl\xa1\xca\xe3\xa2\xf0:\xd7\x00\xc9g\x02f1\xbd?\xb3\xc4\xe10S7\x9d\x91\x84\xc9y\xb2V<\x7f\x87\x14\xde\xc2\xf4\x94\xabyk\xc1\xbcU\x14\xcb\xb8\xcf\x96\xfedU\x81Z\xff\t\xa6\xe0\xf5\x15p\x91\xf0\xec\x7f')]"]
```
Comment (gpotter2) [inline test/regression.uts]: I think you can remove the previous ones

**Comment by wwencel** [comment]
my previous commit was a mistake, sorry for this. It should be all fine now

**Comment by guedou** [comment]
Could you add unit tests corresponding to the newly added options?

**Comment by wwencel** [comment]
sorry for not getting back to the subject, we had pretty intense time. 

is this one unit test containing each type of new options is sufficient? Or should I extend it more?

File: test/regression.uts [hunk 3]
```diff
@@ -11455,9 +11455,8 @@ assert BOOTP().answers(BOOTP())
 assert BOOTP().hashret() == b"\x00\x00\x00\x00"
 
 import random
-random.seed(0x2807)
```
Comment (guedou) [inline test/regression.uts]: Could you keep this test?

File: test/regression.uts [hunk 4]
```diff
@@ -11476,15 +11475,20 @@ assert udof.m2i("", unknown_value_end) == [(254, b'\xff'*255), 'end']
 unknown_value_pad = b"\xfe" + b"\xff"*256 + b"\x00"
 udof = DHCPOptionsField("options", unknown_value_pad)
 assert udof.m2i("", unknown_value_pad) == [(254, b'\xff'*255), 'pad']
-
```
Comment (guedou) [inline test/regression.uts]: Please do not remove this line.

File: test/regression.uts (hunk 3 above)
Comment (wwencel) [reply test/regression.uts]: I can, but number of options has changed, so it's selecting different option than before and with 0x2807 seed value is:
[('v4-captive-portal', b'$#\xb5\xb7\xd0\xbf \xe2\xb8\x88\x19\xffs\x99]\xe6@\xb9T\xaa\xcfV\xd9\xe8\xa8\x19uE\x01\xcf\xe3\x04\x88\xccb\x19~\x9e\xe3a\xa7\x17o\xbe\xa2\xf2\x84\xadx\xfe\xff)\x0fH\xf3\xf6\xd6\xb0\xa7A\xaa\xbb((,\xc8\x13\xaap\xb5\xef\x12\xddV*\xe0\xffnWSA\xa4\x88\x83\xd7\xf5\x83\x92K\xa3\xf1F}\x81B[\xc5\xb1\xb7\xb2\xff\xd0\xb8\x10\xcf\xadZo\xb4\xf4\x99c\xf4\xfe\xae\x82 TV#\x05\xdd\xbc\xf7\xb8%\x8dr\xb8\xc0\xdb"\x05\xf4\xc1\x9ea\x83%\x86\xf5\xdd\xe8r\x03\x0f\x11\x1c\x7f|\xcdm\xc7v\x16{T\xe2\x81\xdd\xf9W|\xb7U=\xd5\xe0p\xdf\xd9\x94\xcf\x88Sy\x86\r\xb7\xb0\xc5=,\xe0\xaa\x05\xb8\xf5\xe8>\xa3nl\xf7Y\x0f\xfd\xe5D\xea\xd7\xffc\xfcP\x0f6Q\xd4\x9f\x82\xb6\xda\xd4t\xa3\x18\xe6\xdeKD\xf8t\xa5\xb1%O\xed\xad?k\xa8\xffNq\x88\xec\xac{.\\\xfc\xf1\x12t\xd9=+\xc5qO\xf7\xf4\x14\x07pY\xfa\x9e\x7f\x1b\xf4}\x04\xc9r\x00\xbcU\x91\xa7\xe2I\xbc\xd8kt\xafh\xa3\x14\xa3\xc1\x1d\x93\xfa\x8cH\r\x95C\xc6\xd6G\'}\xb5\x04z\xc7w\xeb\xfes\xf7]\x9fi\'\xcc\xfc\xdd\nbW\xeeq+\xc4\xe5\x89\xe9X\xf6D\xb0P\xc87\xa6Kk\xf0\x98Ct(!\'\xb7e\xde19\xd1\'\xdbc\xc1n)u mf5\xf4f\x93\xa7\t+\x01\xcep\x1d8\xb0\x1b\xc1\x96\x0e\xed\xe4\xb5\x0b^N\x07\xb7\xe1~K\r\t\x8eyr\xb6(\xdd0\xa1(\xdf\xc3y+W8x\x85\xf5)\x84\xf4\x83\xdb\x9d\xd9\x8d\\\x99>lf\xeakw\xe2\xdd\x04Tr\x9f\x8b\x17\xa0e\x1d\xc7\xb2\xe2d\x1c\xd4I\xc3\x11\xd7_\xe3\x83W\xb8\xb4\xec&\xe2\x07g\xa7\x12\x92\x02{\n\x89\x86\x93\xe6/5OL\xceV&wl_s\x0fE\xa3!W\xd9!\xa7Aq\x82\x0c\'\x9dL\xfe\x8fx\xc5\x93\xfeT\xda\xb3\xa4.\xa8\xdb\x06\x1a\x04\xea\xd9\xe8\xd8\x8b"X\xabt<\x0e\x96Q\xd6\x04\xf5?g\xfa\xabI{\x0e\xb0e\xee\xb9\xd2\xcf\x81{5\x91\xa8\xd6\x1d,\x1d\xb8\x91<\xf7\xd2Z;+\xd8\x06\xa5\xaeo\xe2\x951\x9a\xcb\x05\xca\xc0\xe9\xd1\xcc\xd4\xffD\xc4\xcf5\xa0\x8cv\r\x0c\xd1U\x1e\xcc\xe4@\xe5\xf4\x1d\x889\x9f\xee\xec\x1e\xa4RoS\x80+\xcb3\xb1!wlm\xed+\x16L\xd4\xc3r\xe57\xb39\xfe\x99?h\xc8F2\xb5\xc4\x15M\x00\x90\xe7\xaf\xf2\x7f\xb2\x93\x14\xa5\x95\x88\x9dB\xffW\x18c\xd3\x81\xfb^6\x81\x04\xaa\x84\xe7\x12\'\x17\xee\xbe\x06W\xcaj}\xd6\xd1\xe8f\x97uu\xcd\x0c\xb31\x87\xb9\xb39`\xcb\x03V8\x0f"Q\x13\xcc\xcf1\xa28^-\x80\n\xee\xb8h\xf7\x912\x0c\xce\xe5\x8eo\x0b\xe4~\xfa\x99\x8a\xd9\xaf*@\x80`\xda4\xeb\xf3x\xfe\xfa\x8aG\xebi4\xd2\xa1\xfe\x9f\x85%\xc7\xcd6\xfa\tP\xb2%\x07\xa5\x8e\x8e(4D\x1a\xe3\xaf\xd4\x1b\x82\xe7\xc7\xe89A\x9e\xf0\xefY\x99\x81/\n\xbf\xec\xc1\x0e\x9d*L\n?b23\xfc\xe8k+(\x18\x9d\xc0!+~\x8d-\xa6,x\xac\x95\x1a\xe3\xb9\xb7\x9f\xdb*\x14 \xb2\x18\x0f\xa7\x02\xcc\xfb\x97\x84\xc4\xf3d\x80\xc2\xba\x7f\xa7Q5\xb8\xf9\xdf\xd8\x19\x9a\x1c\'\x01\x9f\xb0\xf9&\t\ty;ZQ-Jq\x13B\tx{2\xf3P\xdb\xf7\x0b\x03\xf0qw\x7f#\x0fd\xe4-T<(\xc1\xcb\x8fh\x94zky\x1fi\xba\xee&\x11\xf9\xdb\x19o-\xfb\xf4\x1fi\x05^]\x95\x9fu/n I +C\xec\xaai\xb4e*0\x06M\xcb\x82\xcc\x87+\xe7\xfb\xbb:\x185|=\xe3N*\\\xa7\xf3\xbcu\xdb\xbdN\x82I\xa4\xd5\xd6\x87sbb\x17C<9.qE:\x99\xdd\xe8@\xdf\xdbEZ\x02h?\xc0\xe3dC\xd9\x01EpU\'\x04/aJV\xa8\x13\x93\xa2LZ\xef\x9c\x0c[\x94H\xc3\x9f\xa3#\x84\x1c\xde\x1e\xefl\x97\x1b\xc2\x965\x88\xa2\xab\x13\xec\x0cg\xdf\xc1\xff\xcb\xd9n\x93?\xf0\x1e\x98\xc3-.\x9b&;\xacPt\x0e.\xb2@q\xcc\xd2\x0f\x19\xe3\xba\xc6SaWC\x02\x1c_\xa2\xe9.b\xff\xd9\x06\x15Y\xf0\xda\\\x00\xf7\x82\xb9eg\xd6\t\x19\xea-\xc0\n\xc23\xbe"r\x0b\xa3\x95K\xa0\xdc\t\xc0\xfb\x9e\x9c\x82jR\xf3\xbc\xf4')]

I just wanted to keep it cleaner, this is why I choose seed that is generating more readable data

Comment (guedou) [reply test/regression.uts]: I understand. You can leave it as it is.

File: test/regression.uts (hunk 4 above)
Comment (wwencel) [reply test/regression.uts]: added back :)

**Pull request merged**